packages/taraxa-ui/src/components/BlockTable/BlockTable.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import {
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  Box,
+  TablePagination,
+  TableContainer,
+  ThemeProvider,
+  CssBaseline,
+} from '@mui/material';
+import Label from '../Label';
+import Button from '../Button';
+import {
+  TransactionIcon,
+  GreenRightArrow,
+  GreenCircledCheck,
+  RedCircledCancel,
+  TransactionBlock,
+} from '../Icons';
+import theme from '../theme';
+
+export enum TransactionStatus {
+  SUCCESS = '0x0',
+  FAILURE = '0x1',
+  IN_PROGRESS = '0x',
+}
+
+export interface TransactionData {
+  txHash: string;
+  status: TransactionStatus;
+  timestamp: string;
+  pbftBlock: string;
+  dagLevel: string;
+  dagHash: string;
+  value: string;
+  from: string;
+  gasLimit: string;
+  gas: string;
+  gasPrice: string;
+  to: string;
+  nonce: number;
+  transactionLink: JSX.Element;
+  addressLinkFrom: JSX.Element;
+  addressLinkTo: JSX.Element;
+}
+
+export const statusToLabel = (state: TransactionStatus) => {
```

Comment:
If this is getting extracted into a component this needs to go into utils

Reply:
@farebord I tend to agree with @VargaElod23 . This is more than just a component. It's basically a whole container. Moving this inside the taraxa-ui might be too much. From what I saw in the design, we use this type of component in 3 different pages, like transactions/xxx, block/xxx, address/xxx. We should extract the small components used in all 3 pages, if there are any, and keep this container in the explorer. Because if we need another page which is similar 90% but has a small change, we either need to extend this in taraxa-ui and make sure it doesn't brake in other pages, or create a new component, both would be too much. What do you think?

Reply:
Or, alternatively, make it really reusable with custom table columns and rows. Might be a bit more timeconsuming and not sure how customizable it needs to be.

Reply:
I'll try to make this more customizable and move utils to the explorer. The point is there are no shared utils in taraxa-UI and it seems they should be there already. @Malak67 @VargaElod23

Reply:
Yes, I also looked for it. You can leave it there or create an utils and export it so we can use them in explorer as well.